api/client/client.cc

```diff
@@ -106,23 +117,45 @@ int main(int argc, char **argv) {
         Help();
         return 1;
     }
-
-    string cmd = string(argv[1]);
-    if (cmd == "nic") {
-        return SubNic(argc, argv, options);
-    } else if (cmd == "sg") {
-        return SubSg(argc, argv, options);
-    } else if (cmd == "status") {
-        return sub_status(argc, argv, options);
-    } else if (cmd == "shutdown") {
-        return sub_shutdown(argc, argv, options);
-    } else if (cmd == "request") {
-        return SubRequest(argc, argv, options);
-    } else if (cmd == "dump") {
-        return SubDump(argc, argv, options);
-    } else {
-        cerr << "invalid subcommand " << cmd << endl;
-        Help();
-        return 1;
+    int id_command = 0;
+    int id_insert = 0;
+    std::vector<Cmd> cmds;
+    Cmd command(argv, 0);
+    cmds.push_back(command);
+    for (int i = 0; i < argc; i++) {
+        string cmd = argv[id_insert];
+        if (cmd == "+") {
+            cmds[id_command].size = id_insert;
+            argv += id_insert;
+            id_command++;
+            Cmd command(argv, 0);
+            cmds.push_back(command);
+            id_insert = 1;
+        } else {
+            id_insert++;
+        }
+    }
+    cmds[id_command].size = id_insert;
+    int ret = 0;
+    for (unsigned int i = 0; i < cmds.size(); i++) {
+        string cmd = cmds[i].command[1];
+        if (cmd == "nic") {
+            ret = SubNic(cmds.at(i).size, cmds.at(i).command, options);
```

Comment:
here you are assigning ret to the return of SubNic.
and by doing so you are erasing it's old value too